lib/lambda/cloudwatch-custom-widget/readers.mts

```diff
@@ -1606,87 +1584,87 @@ function parseReplicationFactor(
   return factors;
 }
 
-interface RestatectlSqlOutput {
-  headers: Map<string, number>;
-  rows: string[][];
+interface PartitionState {
+  partition_id: number;
+  plain_node_id: string;
+  gen_node_id: string;
+  target_mode: string;
+  effective_mode: string;
+  updated_at: string;
+  leader_epoch: number;
+  leader: string;
+  applied_log_lsn: number;
+  last_record_applied_at: string;
+  replay_status: string;
+  durable_log_lsn: number;
+  archived_log_lsn?: number;
+  target_tail_lsn: number;
+}
```

Comment:
Given PartitionState is defined here, as this specific ordered sequence of fields, then I think it would make sense to  SELECT these same fields, rather than `*`, in the corresponding SQL statement, which ideally could be defined as a string/const/whatever next to this interface? (Same for NodeState.)

Reply:
Love the suggestion, makes a ton of sense! Please can I do defer this as a follow-up?

Reply:
the only caveat is that in the current impl, you can default/set undefined a field that isnt in the * output, whereas if we name the fields specifically, then the sql query will fail if the field is removed. i think i used some * originally so we could more gracefully cover that removed field case

Reply:
Yeah, that's exactly it; with the current approach a column gone missing field will still render as `undefined`. I think we'd prefer this given the user can deploy an arbitrary version of restate.